applet/src/main/java/applet/SchnorrZKP.java

```diff
@@ -1,16 +1,10 @@
 package applet;
 
-import org.bouncycastle.math.ec.ECPoint;
 import java.math.BigInteger;
+import org.bouncycastle.math.ec.ECPoint;
 
 public class SchnorrZKP {
 
-    // (G in doc) a base point on the curve that serves as a generator
-    private ECPoint generator;
-
-    // (h in doc) the coFactor of the subgroup, which is usually a small integer up to 4
-    private BigInteger coFactor;
-
```

Comment:
Why is `generator` and `coFactor` not needed now?

Reply:
@OTristanF

Reply:
- We need the generator but it is turned into a local variable. 
- The cofactor was not used since we can get it by generator.getCurve().getCofactor();
- As I look at it, we don't need the prime order either, since it is also included – generator.getCurve().getOrder();